core/src/main/scala/org/bitcoins/core/crypto/TxSigComponent.scala

```diff
@@ -44,6 +46,62 @@ sealed abstract class TxSigComponent {
 
 object TxSigComponent {
 
+  private def inputIndex(inputInfo: InputInfo, tx: Transaction): UInt32 = {
+    tx.inputs.zipWithIndex
+      .find(_._1.previousOutput == inputInfo.outPoint) match {
+      case Some((_, index)) => UInt32(index)
+      case None =>
+        throw new IllegalArgumentException(
+          "Transaction did not contain expected input.")
```

Comment:
put the outpoint in the error message, let's try to give useful error messages with input given by the user :-)